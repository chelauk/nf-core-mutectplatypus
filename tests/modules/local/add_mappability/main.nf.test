nextflow_process {

    options "-dump-channels"
    name "Test Process ADD_MAPPABILITY"
    script "modules/local/add_mappability/main.nf"
    process "ADD_MAPPABILITY"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // tuple path(mappability), val(patient), val(id), path(maf)
                input[0] = ["/Users/chela.james/sandbox/nextflow/test_run/test_mappability.out",
                            'test_patient',
                            'sample',
                            "/Users/chela.james/sandbox/nextflow/test_run/test_patient_sample.maf"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
