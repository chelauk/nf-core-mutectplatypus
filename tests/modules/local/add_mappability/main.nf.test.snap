{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "test_patient",
                        "sample",
                        "test_patient_sample.adj.maf:md5,22fc4d2789c22649822aa2068f68d5aa"
                    ]
                ],
                "fixed_maf": [
                    [
                        "test_patient",
                        "sample",
                        "test_patient_sample.adj.maf:md5,22fc4d2789c22649822aa2068f68d5aa"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T11:19:49.139114"
    }
}